day-4/contract-tracker/src/App.js

```diff
@@ -113,31 +128,101 @@ class App extends Component {
     super(props);
 
     this.state = {
-      contractList: []
+      contractList: [],
+      editIndex: -1,
+      editContract: {title: "", description: "", price: ""}
     };
+
+    this.createContract = this.createContract.bind(this);
+    this.uploadContract = this.uploadContract.bind(this);
+    this.editContract = this.editContract.bind(this);
   }
 
-  onSubmit (contact) {
-    this.state.contractList[this.state.contractList.length] = contact;
+
+  uploadContract (editIndex) {
+    this.setState({
+      ...this.state,
+      editIndex: editIndex,
+      editContract: this.state.contractList[editIndex]
+    });
+  }
+
+  editContract (title, description, price, index) {
+    let newContract = {
+      title: title,
+      description: description, 
+      price: price
+    };
+
+    console.log(newContract);
+    console.log(index);
```

Comment:
these are more than fine for development purposes, but need to be removed before getting merged into develop